### DISCUSSION TITLE
Is loading untrusted html/javascript in VSCode webview safe?

### DISCUSSION THREAD

**Discussion by fabioz** [microsoft/vscode-discussions#366] [Extension Development QnA] [answered]
I'd like to show HTML generated from an external tool in a VSCode Webview.

Is that safe in general or does that mean that some vulnerability could be exploited in this case if malicious javascript is present in the HTML? Is some special setup needed?

Note: the related HTML needs javascript, so, it's not possible to set enableScripts: false it in this case.

**Comment by alefragnani** [comment, marked as answer]
Hi,

_As with any webpage, when creating a webview, you must follow some basic security best practices._ 
(https://code.visualstudio.com/api/extension-guides/webview#security)

The VS Code website has a special section about Webviews on its [Extension Guides](https://code.visualstudio.com/api/extension-guides/webview), so I suggest you to take a deeper look on it. Specially the [Loading local content](https://code.visualstudio.com/api/extension-guides/webview#loading-local-content) and [Security](https://code.visualstudio.com/api/extension-guides/webview#security) topics. 

Hope this helps
